packages/expo-location/android/src/main/java/expo/modules/location/records/LocationArguments.kt

```diff
@@ -45,16 +45,24 @@ internal class LocationTaskOptions(
   @Field var deferredUpdatesTimeout: Float? = null,
   @Field var foregroundService: LocationTaskServiceOptions? = null
 ) : LocationOptions() {
-  internal fun toMutableMap() = mutableMapOf(
-    "accuracy" to accuracy,
-    "distanceInterval" to distanceInterval,
-    "mayShowUserSettingsDialog" to mayShowUserSettingsDialog,
-    "timeInterval" to timeInterval,
-    "deferredUpdatesDistance" to deferredUpdatesDistance,
-    "deferredUpdatesInterval" to deferredUpdatesInterval,
-    "deferredUpdatesTimeout" to deferredUpdatesTimeout,
-    "foregroundService" to (foregroundService?.toMutableMap() ?: mutableMapOf())
-  )
+  internal fun toMutableMap(): MutableMap<String, Any?> {
+    val map: MutableMap<String, Any?> = mutableMapOf(
+      "accuracy" to accuracy,
+      "distanceInterval" to distanceInterval,
+      "mayShowUserSettingsDialog" to mayShowUserSettingsDialog,
+      "timeInterval" to timeInterval,
+      "deferredUpdatesDistance" to deferredUpdatesDistance,
+      "deferredUpdatesInterval" to deferredUpdatesInterval,
+      "deferredUpdatesTimeout" to deferredUpdatesTimeout
+    )
+
+    // Only add foregroundService to the map if it's not null
+    foregroundService?.let {
+      map["foregroundService"] = (foregroundService?.toMutableMap() ?: mutableMapOf()) as Any
+    }
+
+    return map
+  }
```

Comment:
I think this `?: mutableMapOf()` shouldn't be necessary as we know that the foreground service is not null. We should be able to safely use something like this instead:

```suggestion
internal fun toMutableMap(): MutableMap<String, Any?> = mutableMapOf<String, Any?>(
    "accuracy" to accuracy,
    "distanceInterval" to distanceInterval,
    "mayShowUserSettingsDialog" to mayShowUserSettingsDialog,
    "timeInterval" to timeInterval,
    "deferredUpdatesDistance" to deferredUpdatesDistance,
    "deferredUpdatesInterval" to deferredUpdatesInterval,
    "deferredUpdatesTimeout" to deferredUpdatesTimeout
  ).apply {
    foregroundService?.let {
      this["foregroundService"] = it.toMutableMap()
    }
  }
```